pkg/crds/provisioningv2/crds.go

```diff
@@ -21,6 +21,7 @@ var (
 		"MachineDeployment":  true,
 		"MachineSet":         true,
 		"Cluster":            true,
+		"MachinePool":        true,
```

Comment:
Let's add a comment explaining why this is needed with a link to upstream issue.
@mitulshah-suse / @chiukapoor , please handle opening `r/r` issue for removing the workaround - hopefully we get fixed CAPI release before we can't bump it for v2.11.0 release and won't need to worry about migrations.